properly pass compressionType and requiredAcks to BackbeatProducer

welansari opened the pull request:
Issue: BB-620

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-620 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.7.0`


Please check the `Fix Version/s` of BB-620, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented on extensions/notification/destination/KafkaNotificationDestination.js:
```diff
@@ -52,6 +52,8 @@ class KafkaNotificationDestination extends NotificationDestination {
             pollIntervalMs,
             auth,
+            compressionType: 'none',
+            requiredAcks: '1',
```
these params may need to be added to the destination config, may be done now or later, but need ticket either way (at product or at least zenko level, so that we can also track the remaining work: zkop, ...)

francoisferrand commented on lib/BackbeatProducer.js:
```diff
@@ -57,7 +57,7 @@ class BackbeatProducer extends EventEmitter {
                 maxRequestSize: joi.number().default(KAFKA_PRODUCER_MESSAGE_MAX_BYTES),
                 compressionType: joi.string().default(KAFKA_PRODUCER_DEFAULT_COMPRESSION_TYPE),
-                requiredAcks: joi.number().default(KAFKA_PRODUCER_DEFAULT_REQUIRED_ACKS),
+                requiredAcks: joi.string().default(KAFKA_PRODUCER_DEFAULT_REQUIRED_ACKS),
```
why pass a string? this reduces the effectiveness of validation, as any string will be valid...

if we just neet to support some way of passing "empty" param to kafka, i.e. use kafka defaults, we should be able to just do `joi.number().allow("").default(KAFKA_PRODUCER_DEFAULT_REQUIRED_ACKS)` : which (according to the docs) should allow numbers or empty string, and fall back to default value if `undefined` value is provided

(or maybe just `joi.string().regex(/^[0-9]*$/, 'number').default(KAFKA_PRODUCER_DEFAULT_REQUIRED_ACKS)`)

welansari replied:
I passed a string because the default value used in Artesca is 'all' which waits for all brokers to ack. Looks like we can use -1 too for the same purpose, Leaving it as a number.

francoisferrand commented on extensions/gc/GarbageCollector.js:
```diff
@@ -132,6 +132,8 @@ class GarbageCollector extends EventEmitter {
                 hosts: this._kafkaConfig.hosts,
                 site: this._kafkaConfig.site,
+                compressionType: this._kafkaConfig.compressionType,
+                requiredAcks: this._kafkaConfig.requiredAcks,
```
won't this change the behavior on artesca?

If there are some good values, best to use them indeed;
but not sure if either
- we should rely on defaults from Kafka : maybe they are good (→ should be used as default everywhere, i.e. both s3c and zenko) or maybe we have better values (→ should be set in both products)
- we should actually have default values in backbeat, if both product don't agree
- if the products ahve different defaults, should probably be explicit : i.e. put the values in each product (or if we can define a "correct" default, but don't want to change it in S3C for compatibility: maybe defaut value in backbeat and "override" the default in s3c)

welansari replied:
The default values is what we use in Artesca, so no the behaviour won't change:
- Waiting for all brokers to ack the messages
- Using `Zstd` compression

I've talked with the S3C team and for RING 9.x we want to keep the same config from before :
- Only wait for one broker to ack the messages.
- Use broker level compression, no compression done by the producers.
The custom values will be set in the backbeat config in federation

francoisferrand commented on lib/config.joi.js:
```diff
@@ -33,7 +33,7 @@ const joiSchema = joi.object({
         site: joi.string(),
         compressionType: joi.string().default(KAFKA_PRODUCER_DEFAULT_COMPRESSION_TYPE),
-        requiredAcks: joi.number().default(KAFKA_PRODUCER_DEFAULT_REQUIRED_ACKS),
+        requiredAcks: joi.string().default(KAFKA_PRODUCER_DEFAULT_REQUIRED_ACKS),
```
same

Commit:
allow passing compressionType and requiredAcks for Kafka notification destinations

Issue: BB-620

Commit:
properly pass compressionType and requiredAcks to BackbeatProducer

Issue: BB-620

benzekrimaha reviewed (approved):
LGTM

Commit:
revert kafka consumer lag PR

RING 9.4 now deploys a kafka-exporter, this should be used
to export metrics instead of manually doing it in backbeat
containers

Code is from BB-561

Issue: BB-574

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-620 contains:


* `8.7.0`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `9.0.0`


Please check the `Fix Version/s` of BB-620, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

Commit:
set default value for replicationGroupId

Issue: BB-620

Commit:
allow empty probe config when no replication endpoint defined

Data processors are still deployed in S3C when no replication
endpoint is configured.

Issue: BB-620

Commit:
fix backbeatAPI always expecting ingestion config

S3C does not support ingestion

Issue: BB-620

Commit:
make s3 config optional

The S3 config is not needed in S3C when only
deploying bucket notifications

Issue: BB-620

bert-e commented:
# Conflict




There is a conflict between your branch `bugfix/BB-620` and the
destination branch `development/9.0`.

Please resolve the conflict on **the feature branch** (`bugfix/BB-620`).

```sh
git fetch && \
git checkout origin/bugfix/BB-620 && \
git merge origin/development/9.0
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:bugfix/BB-620
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

welansari commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/9.0`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`
* `development/8.6`



Please check the status of the associated issue BB-620.

Goodbye kerkesni.




*The following options are set:* **approve**
